src/main/java/com/empowerops/linqalike/CommonDelegates.java

```diff
@@ -32,7 +33,7 @@ public boolean equals(Object left, Object right) {
 
                 @Override
                 public int hashCode(Object object) {
-                    counter = (counter + 1) % Integer.MAX_VALUE; //mod prevents integer overflow
+                    counter = (counter % Integer.MAX_VALUE) + 1; //mod prevents integer overflow
```

Comment:
actually, nuts to this, lets just use the System hashcode, its fast & well tested, and when we have a 'collision', we'll require the further call to 'equals'.